flink-runtime/src/test/java/org/apache/flink/runtime/rest/RestClientSslTest.java

```diff
@@ -0,0 +1,315 @@
+import org.apache.flink.shaded.netty4.io.netty.channel.socket.SocketChannel;
+import org.apache.flink.shaded.netty4.io.netty.channel.socket.nio.NioServerSocketChannel;
+import org.apache.flink.shaded.netty4.io.netty.handler.codec.http.DefaultFullHttpResponse;
+import org.apache.flink.shaded.netty4.io.netty.handler.codec.http.FullHttpResponse;
+import org.apache.flink.shaded.netty4.io.netty.handler.codec.http.HttpContent;
+import org.apache.flink.shaded.netty4.io.netty.handler.codec.http.HttpHeaderNames;
+import org.apache.flink.shaded.netty4.io.netty.handler.codec.http.HttpRequest;
+import org.apache.flink.shaded.netty4.io.netty.handler.codec.http.HttpRequestDecoder;
+import org.apache.flink.shaded.netty4.io.netty.handler.codec.http.HttpResponseEncoder;
+import org.apache.flink.shaded.netty4.io.netty.handler.codec.http.HttpResponseStatus;
+import org.apache.flink.shaded.netty4.io.netty.handler.ssl.ClientAuth;
+import org.apache.flink.shaded.netty4.io.netty.handler.ssl.SniHandler;
+import org.apache.flink.shaded.netty4.io.netty.handler.ssl.SslContext;
+import org.apache.flink.shaded.netty4.io.netty.handler.ssl.SslContextBuilder;
+import org.apache.flink.shaded.netty4.io.netty.handler.ssl.util.InsecureTrustManagerFactory;
+import org.apache.flink.shaded.netty4.io.netty.util.CharsetUtil;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.io.File;
+import java.io.InputStream;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.nio.file.Files;
+import java.security.KeyStore;
+import java.security.PrivateKey;
+import java.security.cert.X509Certificate;
+import java.util.List;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
+import static org.apache.flink.shaded.netty4.io.netty.handler.codec.http.HttpResponseStatus.OK;
+import static org.apache.flink.shaded.netty4.io.netty.handler.codec.http.HttpVersion.HTTP_1_1;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+/** Ssl cases for {@link RestClient} and {@link RestServerEndpoint}. */
+public class RestClientSslTest extends TestLogger {
+
+    private static final Time timeout = Time.seconds(10L);
+
+    private static final String TRUST_STORE_PATH =
+            RestClientSslTest.class.getResource("/local127.truststore").getFile();
+
+    private static final String KEY_STORE_PATH =
+            RestClientSslTest.class.getResource("/local127.keystore").getFile();
+
+    private static final String HOST_SSL = "flink.local";
+
+    private static final NioEventLoopGroup bossGroup = new NioEventLoopGroup(1);
+    private static final NioEventLoopGroup workerGroup = new NioEventLoopGroup(1);
+
+    private final Configuration config;
+
+    private RestClient restClient;
+
+    private ServerBootstrap serverEndpoint;
+
+    private static String sniHostReceived;
+
+    static {
```

Comment:
This static magic can have serious effect on other tests so just must be solved in a different way.